clevertap-core/src/main/java/com/clevertap/android/sdk/Utils.java

```diff
@@ -564,4 +525,43 @@ public static boolean isMainProcess(Context context, String mainProcessName) {
         }
         return false;
     }
+
+    public static List<JSONObject> toJSONObjectList(JSONArray jsonArray) throws JSONException {
+        List<JSONObject> jsonObjectList = new ArrayList<>();
+        for (int index = 0; index < jsonArray.length(); index++) {
+            jsonObjectList.add(jsonArray.getJSONObject(index));
+        }
+        return jsonObjectList;
+    }
+
+    /**
+     * Calculates the haversine distance between two locations on Earth.
+     *
+     * @param coordinateA The first location.
+     * @param coordinateB The second location.
+     * @return The haversine distance between the two locations, in kilometers.
+     */
+    public static double haversineDistance(Location coordinateA, Location coordinateB) {
```

Comment:
push in new utils class. and mentioned source from where we picked this

Reply:
@shivamsharma2710